File: html/redcap_v15.7.6/API/examples/perl/import_events.pl
#!/usr/bin/env perl

use strict;
use warnings;
use LWP::Curl;
use JSON;

my %config = do 'config.pl';

my $record = {
	event_name        => 'Event 1',
	arm_num           => 1,
	day_offset        => 0,
	offset_min        => 0,
	offset_max        => 0,
	unique_event_name => 'event_1_arm_1',
};

my $json = JSON->new->allow_nonref;
my $data = $json->encode([$record]);

my $fields = {
    token    => $config{api_token},
    content  => 'event',
	action   => 'import',
    format   => 'json',
	override => 0,
    data     => $data,
};

my $ch = LWP::Curl->new(auto_encode => 0);
my $content = $ch->post($config{api_url}, $fields, $config{referer});

print $content;

File: html/redcap_v15.7.6/API/examples/r/import_events.R
#!/usr/bin/env Rscript

source('config.R')
library(RCurl)
library(jsonlite)

record <- c(
	event_name='Event 1',
	arm_num=1,
	day_offset=0,
	offset_min=0,
	offset_max=0,
	unique_event_name='event_1_arm_1'
)

data <- toJSON(list(as.list(record)), auto_unbox=TRUE)

result <- postForm(
    api_url,
    token=api_token,
    content='event',
	action='import',
    format='json',
	override=0,
    data=data
)
print(result)
